### DISCUSSION TITLE
Cloudflare R2

### DISCUSSION THREAD

**Discussion by reavpC6K** [saleor/saleor#9953] [Ideas]
https://blog.cloudflare.com/introducing-r2-object-storage/

Cloudflare launched R2 which includes full S3 API compatibility. Unlike S3 it has cheaper storage and free egress.
How can S3 be swapped in self-hosted Saleor?

**Comment by heiko-r** [comment]
It's a bit late, but in case anyone else wonders how to set up S3 compatible storage:
There are a few more environment variables than mentioned in the documentation. See `saleor/settings.py` for the whole list. You can set the S3 compatible endpoint URL using the `AWS_S3_ENDPOINT_URL` variable.

**Reply by cybars69** [reply]
A bit late to ask, but did you manage to get it running with R2 storage?

**Reply by heiko-r** [reply]
I didn't try R2, just Alibaba Cloud's OSS. But that was all long ago with Saleor 2, can't help with the current version at all.
